src/Workspaces/Core/Portable/ObsoleteSymbol/AbstractObsoleteSymbolService.cs

```diff
@@ -31,16 +31,14 @@ public async Task<ImmutableArray<TextSpan>> GetLocationsAsync(Document document,
     {
         var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
 
-        var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
>GetRequiredSemanticModelAsync [](http://example.com/codeflow?start=43&length=29)

Should we audit other uses of GetSemanticModel that could use the member off the Document instead?

Reply:
Yes.  Please do :)